test/unit/org/apache/cassandra/db/compaction/CompactionsTest.java

```diff
@@ -533,6 +548,104 @@ private void testDontPurgeAccidentaly(String k, String cfname) throws Interrupte
         assertTrue("should be empty: " + cf, cf == null || !cf.hasColumns());
     }
 
+    @Test
+    public void incompletedCompactionAbortNotRemovedFromCompactionsInProgress() throws IOException
+    {
+        Keyspace keyspace = Keyspace.open(KEYSPACE1);
+        String cfName = CF_STANDARD5;
+        ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(cfName);
+
+        cfs.clearUnsafe();
+        cfs.disableAutoCompaction();
+
+        SchemaLoader.insertData(KEYSPACE1, cfName, 0, 1);
+        cfs.forceBlockingFlush();
+
+        SSTableReader s = SSTableRewriterTest.writeFile(cfs, 1000);
+        cfs.addSSTable(s);
+        SSTableReader s2 = SSTableRewriterTest.writeFile(cfs, 1000);
+        cfs.addSSTable(s2);
+
+        assertEquals(3, cfs.getSSTables().size());
+
+        Set<SSTableReader> compacting = Sets.newHashSet(s, s2);
+        LifecycleTransaction txn = cfs.getTracker().tryModify(compacting, OperationType.UNKNOWN);
+        CompactionTask compaction = spy(new FailedAbortCompactionTask(cfs, txn, 0, CompactionManager.NO_GC, 1024 * 1024, true));
+        try
+        {
+            compaction.runMayThrow();
+        }
+        catch (Throwable e)
+        {
+            assertNotNull(e);
```

Comment:
Assert exception is what you expect as well? Either by adding a message and/or checking the suppressed exception?